src/main/java/uk/gov/justice/digital/delius/service/AttendanceService.java

```diff
@@ -0,0 +1,54 @@
+package uk.gov.justice.digital.delius.service;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.digital.delius.data.api.Attendance;
+import uk.gov.justice.digital.delius.data.api.Attendance.ContactTypeDetail;
+import uk.gov.justice.digital.delius.jpa.standard.entity.Contact;
+import uk.gov.justice.digital.delius.jpa.standard.repository.ContactRepository;
+
+@Slf4j
+@Service
+public class AttendanceService {
+
+    private final ContactRepository contactRepository;
+
+    @Autowired
+    public AttendanceService(final ContactRepository contactRepository) {
+        this.contactRepository = contactRepository;
+    }
+
+    public Optional<List<Contact>> getContactsForEvent(final Long offenderId, final Long eventId, final LocalDate localDate) {
+        final List<Contact> contacts = contactRepository.findByOffenderAndEventIdEnforcement(offenderId, eventId, localDate);
+        return contacts.isEmpty() ? Optional.empty() : Optional.of(contacts);
+    }
+
+    static boolean forEntityBoolean(final String booleanStr) {
+        if (booleanStr == null)
+            return false;
+        return booleanStr.trim().equals("1");
+    }
+
+    public static List<Attendance> attendancesFor(List<Contact> contacts) {
```

Comment:
consider using:

1. the existing TypesTransformer instead of your own forEntityBoolean
2. moving code to a transformer class and testing separately

Reply:
I have used TypeTransformer myself elsewhere but didn't see where / how I would do this. Given that there was at least one other wrapper type class, I thought this was OK

Reply:
Didn't know there was an existing types transformer. TBH everywhere else I have worked the entity object would be mapped as a Boolean or boolean. Working with a flag field which is mapped to a String in a JPA object is not what I'd like to see in the first place. If I am to do something different I'd like to fix the JPA CONTACT entity but that seemed like it would hit other classes